node/components/bft/src/chonky_bft/mod.rs

```diff
@@ -276,7 +275,14 @@ impl StateMachine {
         ctx: &ctx::Ctx,
         qc: &validator::CommitQC,
     ) -> ctx::Result<()> {
-        self.high_commit_qc = max(Some(qc.clone()), self.high_commit_qc.clone());
+        if self
+            .high_commit_qc
+            .as_ref()
+            .map_or(false, |old| old.view().number >= qc.view().number)
```

Comment:
```suggestion
            .map_or(false, |old| old.view().number > qc.view().number)
```
Nit: Greater than should be enough. If we have two QCs at the same view, they must be for the same block.

Reply:
if old view == qc view, there is nothing to be done, so IMO this WAI